### PR TITLE
dapps-precompile changes for polkadot-v0.9.16

### PR DESCRIPTION
**Pull request by Maar-io**
**Pull Request Summary**

> update pallet to support polkadot-v0.9.16

**Check list**
- [ ] contains breaking changes
- [ ] adds new feature
- [ ] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tests and/or benchmarks are included
- [ ] changed API client type definition or chain metadata

**This pull request makes the following changes:**

**Adds**
- (ex: Add feature A)

**Fixes**
- (ex: Fix validation function)

**Changes**
- (ex: Change document B)

**To-dos**
> *Feel free to remove this section if it's not applicable

- [ ] (ex: add user list)

### CODE REVIEW

File: precompiles/dapps-staking/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    "scale-info/std",
+    "evm/std",
+    "sp-std/std",
+    "sp-core/std",
+    "sp-runtime/std",
+    "fp-evm/std",
+	"frame-support/std",
+	"frame-system/std",
+    "pallet-dapps-staking/std",
+    "pallet-evm/std",
```
Comment (Dinonard) [inline precompiles/dapps-staking/Cargo.toml]: Please fix the formatting

File: precompiles/dapps-staking/src/mock.rs [hunk 2]
```diff
@@ -81,13 +81,14 @@ impl Default for TestAccount {
...
-    fn into_account_id(h160_account: H160) -> TestAccount {
+// needed for associated type in pallet_evm
+impl AddressMapping<AccountId> for TestAccount {
+    fn into_account_id(h160_account: H160) -> AccountId {
         match h160_account {
-            a if a == H160::repeat_byte(0x11) => Self::Alex,
-            a if a == H160::repeat_byte(0x22) => Self::Bobo,
-            a if a == H160::repeat_byte(0x33) => Self::Dino,
-            _ => Self::Empty,
+            a if a == H160::repeat_byte(0x11) => TestAccount::Alex as u64,
```
Comment (Dinonard) [inline precompiles/dapps-staking/src/mock.rs]: I'd suggest you keep it consistent with the `AccountId` type. Just call TestAccount::Alex.into();

**Review by Dinonard — APPROVED** [review]
LGTM

File: precompiles/dapps-staking/Cargo.toml (hunk 1 above)
Comment (Maar-io) [reply precompiles/dapps-staking/Cargo.toml]: fixed

File: precompiles/dapps-staking/src/mock.rs (hunk 2 above)
Comment (Maar-io) [reply precompiles/dapps-staking/src/mock.rs]: thanks, changed!

**Pull request merged**